### PR TITLE
AWS deployment

### LINKED ISSUE #30: AWS CloudFormation template
Implement a skeleton CloudFormation template to allow us to easily build and tear our stack. This stack should use the AWS Lambda / AWS API Gateway as we only need 1 endpoint on the cloud, which is: GET dates/{day}/images

### PR DESCRIPTION
**Pull request by faisal-fawad**
Details:
- AWS CloudFormation templates for deploy, billing and roles
- Scuffed `build_lambda_code.py` that builds a .zip file for deployment
- GitHub actions deploy pipeline which utilizes scripts in `aws` directory

NOTE: The bandaid mentioned here is simply how the pipeline can be invoked - since `on: [workflow_dispatch]` means the pipeline can only be manually triggered on the default branch, we need `on: [..., pull_request]` to test our changes on a feature branch

### CODE REVIEW

**Comment by faisal-fawad** [comment]
Will squash and merge once approved LOL. See comment in PR for "bandaid"

File: aws/billing.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          Subscribers:
+            - SubscriptionType: EMAIL
+              Address: !Ref emailAddress
+        - Notification:
+            NotificationType: ACTUAL
+            ComparisonOperator: GREATER_THAN
+            Threshold: 100 # 100% of budget is used
+          Subscribers:
+            - SubscriptionType: EMAIL
+              Address: !Ref emailAddress
```
Comment (zachale) [inline aws/billing.yml:95]: This makes me curious if rate limiting is something we can easily add, just to cover our bases

**Review by zachale — APPROVED** [review]
LGTM

**Review by nathan-probert — APPROVED** [review]

File: aws/billing.yml (hunk 1 above)
Comment (faisal-fawad) [reply aws/billing.yml:95]: Will be moved to later PR

**Pull request merged**